src/node/desktop/src/ui/utils.ts

```diff
@@ -1,35 +1,40 @@
+/* eslint-disable max-len */
 const isLocalStorageItemSet = (key: string) => {
-  const value = window.localStorage.getItem(key);
+  if (typeof window !== 'undefined') {
+    const value = window.localStorage.getItem(key);
 
-  return { isSet: value !== '' && !!value, value };
+    return { isSet: value !== '' && !!value, value };
+  }
+
+  return { isSet: false, value: '' };
 };
 
 export const checkForNewLanguage = () => {
-  return new Promise((resolve, reject) => {
-    const now = new Date().getTime();
+  return new Promise((resolve) => {
+    const fisrtRunTime = new Date().getTime();
     const localeStorageItemKey = 'LOCALE';
     const localeLastTimeSetStorageItemKey = 'LAST_TIME';
 
+    // This will check every 0.1 seconds if a new language has been set to the local storage
     const isThereNewLanguageInterval = setInterval(() => {
-      const localeLastTimeData = isLocalStorageItemSet(localeLastTimeSetStorageItemKey);
-
-      if (localeLastTimeData.isSet) {
-        const localeData = isLocalStorageItemSet(localeStorageItemKey);
-
-        if (localeData.isSet && now > parseInt('' + localeLastTimeData.value, 10)) {
-          clearInterval(isThereNewLanguageInterval);
-
-          const newLanguage = localeData.value;
-
-          resolve('' + newLanguage);
+      // After 5 seconds, the default language will be set
```

Comment:
Isn't this more like "give up after 5 seconds"?

Reply:
Yup!